### PR TITLE
Update SignUpServiceUpdater retries exhausted log

### PR DESCRIPTION
**Pull request by rileyanderson**
## Summary
- Update log to include necessary info to pass along to SuS if there are failures
- adds `agreement_version`, `response` (accepted or declined), and `response_time`
- Log looks like - 
  ```ruby
  Rails -- [TermsOfUse][SignUpServiceUpdaterJob] retries exhausted -- { :icn => "icn", :agreement_version => "v1", :attr_package_key => "some-key", :response => "accepted", :response_time => "2024-06-07T15:59:05Z", :exception_message => "some-error" }
  ```

## Testing done
- nothing to really test. updated unit tests

## What areas of the site does it impact?
Terms of use

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

**Review by asg5704 — APPROVED** [review]

File: app/sidekiq/terms_of_use/sign_up_service_updater_job.rb [hunk 1]
```diff
@@ -13,8 +13,23 @@ class SignUpServiceUpdaterJob
       attr_package_key = job['args'].first
       attrs = Sidekiq::AttrPackage.find(attr_package_key)
 
-      Rails.logger.warn('[TermsOfUse][SignUpServiceUpdaterJob] retries exhausted',
-                        { icn: attrs&.dig(:icn), exception_message: exception.message, attr_package_key: })
+      icn = attrs&.dig(:icn)
+      version = attrs&.dig(:version)
+      agreement = UserAccount.find_by(icn:)&.terms_of_use_agreements&.where(agreement_version: version)&.last
```
Comment (dickdavis) [inline app/sidekiq/terms_of_use/sign_up_service_updater_job.rb:18]: You could refactor this to return the `TermsOfUseAgreement` without the safe navigation operators. I think this query would do the trick.

```ruby
TermsOfUseAgreement.joins(:user_account)
                   .where(user_accounts: { icn: })
                   .where(agreement_version: version)
                   .order(created_at: :desc)
                   .first
```

**Review by dickdavis — CHANGES_REQUESTED** [review]

File: app/sidekiq/terms_of_use/sign_up_service_updater_job.rb [hunk 2]
```diff
@@ -13,8 +13,23 @@ class SignUpServiceUpdaterJob
...
+      response = if agreement.present?
+                   agreement.accepted? ? 'accepted' : 'declined'
+                 end
```
Comment (dickdavis) [inline app/sidekiq/terms_of_use/sign_up_service_updater_job.rb:21]: I think this is an example where the safe navigation operator might be useful. You could probably rewrite is as:

```ruby
response = agreement&.accepted? ? 'accepted' : 'declined'
```

You might also be able to do this instead:

```ruby
response = agreement.present? ? agreement.response : 'declined'
```

I think I like the second one more.

Comment (rileyanderson) [reply app/sidekiq/terms_of_use/sign_up_service_updater_job.rb:21]: i want it to be nil if it doesn't exist

**Review by dickdavis — APPROVED** [review]
LGTM, thanks for making the suggested change!

**Review by dickdavis — APPROVED** [review]
LGTM

**Pull request merged**